### PR TITLE
add v2 engine functions placeholder

### PR DESCRIPTION
**Pull request by wjrjerome**

### CODE REVIEW

File: consensus/XDPoS/engines/engine_v2/engine.go [hunk 1]
```diff
@@ -40,3 +42,196 @@ func (consensus *XDPoS_v2) Author(header *types.Header) (common.Address, error)
...
+	// 1. Pull message from the BFTQueue and call the relevant handler by message type, such as vote, timeout or syncInfo
+	// 2. Only 1 message processing at the time
+	return nil
+}
+
+/*
+	SyncInfo workflow
+*/
+// Verify syncInfo and trigger trigger process QC or TC if successful
+func (consensus *XDPoS_v2) VeifySyncInoMessage(header *types.Header) error {
```
Comment (liam-lai) [inline consensus/XDPoS/engines/engine_v2/engine.go]: `VerifySyncInfoMessage`

**Review by liam-lai — APPROVED** [review]

File: consensus/XDPoS/engines/engine_v2/engine.go [hunk 2]
```diff
@@ -40,3 +42,196 @@ func (consensus *XDPoS_v2) Author(header *types.Header) (common.Address, error)
...
+	QC & TC Utils
+*/
+
+// Genrate blockInfo which contains Hash, round and blockNumber and send to queue
+func (consensus *XDPoS_v2) generateBlockInfo() error {
+	return nil
+}
+
+// To be used by different message verification. Verify local DB block info against the received block information(i.e hash, blockNum, round)
+func (consensus *XDPoS_v2) veifyBlockInfo(header *types.Header) error {
```
Comment (wgr523) [inline consensus/XDPoS/engines/engine_v2/engine.go]: veify -> verify

File: consensus/XDPoS/engines/engine_v2/engine.go [hunk 3]
```diff
@@ -40,3 +42,196 @@ func (consensus *XDPoS_v2) Author(header *types.Header) (common.Address, error)
...
+	return nil
+}
+
+func (consensus *XDPoS_v2) veifyQC(header *types.Header) error {
```
Comment (wgr523) [inline consensus/XDPoS/engines/engine_v2/engine.go]: veify -> verify

File: consensus/XDPoS/engines/engine_v2/engine.go [hunk 4]
```diff
@@ -40,3 +42,196 @@ func (consensus *XDPoS_v2) Author(header *types.Header) (common.Address, error)
...
+		1. Verify signer signatures: (List of signatures)
+					- Use ecRecover to get the public key
+					- Use the above public key to find out the xdc address
+					- Use the above xdc address to check against the master node list(For the received QC epoch)
+		2. Verify blockInfo
+	*/
+	return nil
+}
+
+func (consensus *XDPoS_v2) veifyTC(header *types.Header) error {
```
Comment (wgr523) [inline consensus/XDPoS/engines/engine_v2/engine.go]: veify -> verify

File: consensus/XDPoS/engines/engine_v2/engine.go [hunk 5]
```diff
@@ -40,3 +42,196 @@ func (consensus *XDPoS_v2) Author(header *types.Header) (common.Address, error)
...
+
+// Verify round number against node's local round number(Should be equal)
+func (consensus *XDPoS_v2) checkRoundNumber(header *types.Header) error {
+	return nil
+}
+
+// Hot stuff rule to decide whether this node is eligible to vote for the received block
+func (consensus *XDPoS_v2) verifyVotingRule(header *types.Header) error {
+	/*
+		(TODO)
```
Comment (wgr523) [inline consensus/XDPoS/engines/engine_v2/engine.go]: I can write this

Comment (wgr523) [reply consensus/XDPoS/engines/engine_v2/engine.go]: 1. Make sure this node has not voted for this round. We can have a variable `highestVotedRound`, and check `currentRound > highestVotedRound`.
2. HotStuff Voting rule:
- header's round == local current round, AND (one of the following two:)
- header's block extends LockQC's ProposedBlockInfo (we need a isExtending(block_a, block_b) function), OR
- header's QC's ProposedBlockInfo.Round > LockQC's ProposedBlockInfo.Round

**Pull request merged**